src/archive.rs

```diff
@@ -36,4 +38,129 @@ pub struct Header {
     pub terminator: [u8; 2],
 }
 
+/// The fixed length header at the start of a big archive.
+#[derive(Debug, Clone, Copy)]
+#[repr(C)]
+pub struct BigFixedLengthHeader {
+    /// Offset to member table.
+    pub memoff: [u8; 20],
+    /// Offset to global symbol table.
+    pub gstoff: [u8; 20],
+    /// Offset global symbol table for 64-bit objects.
+    pub gst64off: [u8; 20],
+    /// Offset to first archive member.
+    pub fstmoff: [u8; 20],
+    /// Offset to last archive member.
+    pub lstmoff: [u8; 20],
+    /// Offset to last archive member.
+    pub freeoff: [u8; 20],
+}
+
+/// The header at the start of a big archive member.
+#[derive(Debug, Clone, Copy)]
+#[repr(C)]
+pub struct BigHeader {
+    /// File size in decimal.
+    pub size: [u8; 20],
+    /// Next member offset in decimal.
+    pub nxtmem: [u8; 20],
+    /// Previous member offset in decimal.
+    pub prvmem: [u8; 20],
+    /// File modification timestamp in decimal.
+    pub date: [u8; 12],
+    /// User ID in decimal.
+    pub uid: [u8; 12],
+    /// Group ID in decimal.
+    pub gid: [u8; 12],
+    /// File mode in octal.
+    pub mode: [u8; 12],
+    /// Name length in decimal.
+    pub namelen: [u8; 4],
+    // Begining of name, or TERMINATOR.
+    // We deal with it manually.
+    // pub name: [u8; 2]
+}
+
+/// Trait for various header style
+pub trait GenericHeader {
```

Comment:
I think it would make sense to move this trait to `src/read/archive.rs` as it won't be used for archive writing in the future I think.